### PR TITLE
docs: Document `DataFrame.__getitem__` and `Series.__getitem__`

### PR DESCRIPTION
**Pull request by MarcoGorelli**
I was adding docs for these in Narwhals and noticed they don't appear in the Polars docs. Let's add them!

### CODE REVIEW

File: py-polars/docs/source/reference/dataframe/export.rst [hunk 1]
```diff
@@ -8,6 +8,7 @@ Export DataFrame data to other formats:
 .. autosummary::
    :toctree: api/
 
+    DataFrame.__array__
```
Comment (MarcoGorelli) [inline py-polars/docs/source/reference/dataframe/export.rst:11]: drive-by

File: py-polars/polars/lazyframe/engine_config.py [hunk 2]
```diff
@@ -18,7 +18,7 @@ class GPUEngine:
 
     - `device`: Select the device to run the query on.
     - `memory_resource`: Set an RMM memory resource for
-       device-side allocations.
+      device-side allocations.
```
Comment (MarcoGorelli) [inline py-polars/polars/lazyframe/engine_config.py:21]: same

**Comment by alexander-beedie** [comment]
Good idea... 👍 
(FYI: just started looking over the plotting commit ;)

**Review by alexander-beedie — APPROVED** [review]

**Comment by MarcoGorelli** [comment]
cool, thanks for your review! 🙏
